# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieServiceParent.cpp
# Commit: 05d9c3f01472
# Full Hash: 05d9c3f0147288e19525104a250b23f2de1339aa
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-09-16 10:40 +0000
# Regressor Bug: 1982261
# File Overlap Count: 1
# Overlapping Files: netwerk/cookie/CookieService.cpp
# Description:
#   Bug 1982261 - CookieServiceParent logs about invalid cookies instead of crashing, r=valentin,cookie-reviewers
# ==============================================================================

diff -r 157cf44ea0d8 -r 05d9c3f01472 netwerk/cookie/CookieServiceParent.cpp
--- a/netwerk/cookie/CookieServiceParent.cpp	Tue Sep 16 10:23:08 2025 +0000
+++ b/netwerk/cookie/CookieServiceParent.cpp	Tue Sep 16 10:40:49 2025 +0000
@@ -338,10 +338,18 @@
   // update, to make sure we don't send it back to the same content process.
   CookieProcessingGuard guard(this);
 
-  bool ok = mCookieService->SetCookiesFromIPC(aBaseDomain, aOriginAttributes,
-                                              aHost, aFromHttp, aIsThirdParty,
-                                              aCookies, aBrowsingContext);
-  return ok ? IPC_OK() : IPC_FAIL(this, "Invalid cookie received.");
+  nsICookieValidation::ValidationError error =
+      mCookieService->SetCookiesFromIPC(aBaseDomain, aOriginAttributes, aHost,
+                                        aFromHttp, aIsThirdParty, aCookies,
+                                        aBrowsingContext);
+  MOZ_DIAGNOSTIC_ASSERT(error == nsICookieValidation::eOK);
+
+  if (error != nsICookieValidation::eOK) {
+    MOZ_LOG(gCookieLog, LogLevel::Warning,
+            ("Invalid cookie submission from the content process: %d", error));
+  }
+
+  return IPC_OK();
 }
 
 }  // namespace net
